packages/connect-core/src/entities/Repository.ts

```diff
@@ -20,60 +20,63 @@ export interface RepositoryData {
 
 export default class Repository extends CoreEntity implements RepositoryData {
   readonly address!: string
-  readonly author?: string
-  readonly changelogUrl?: string
-  readonly descriptionUrl?: string
-  readonly description?: string
-  readonly environments?: AragonEnvironments
-  readonly icons?: { src: string; sizes: string }[]
+  readonly contentUri?: string
   readonly name!: string
   readonly registry?: string | null
   readonly registryAddress?: string | null
-  readonly roles?: AragonArtifactRole[]
-  readonly screenshots?: { src: string }[]
-  readonly sourceUrl?: string
+  #artifact?: string | null
+  #manifest?: string | null
+  author?: string
+  changelogUrl?: string
+  description?: string
+  descriptionUrl?: string
+  environments?: AragonEnvironments
+  icons?: { src: string; sizes: string }[]
+  roles?: AragonArtifactRole[]
+  screenshots?: { src: string }[]
+  sourceUrl?: string
 
-  constructor(
-    { artifact, manifest, ...data }: RepositoryData,
-    connector: ConnectorInterface
-  ) {
+  constructor(data: RepositoryData, connector: ConnectorInterface) {
     super(connector)
 
-    // TODO: If no metadata, fallback to resolve ourselves with ipfs
-
-    if (artifact) {
-      const { environments, roles }: AragonArtifact = parseMetadata(
-        artifact,
-        'artifact.json'
-      )
-
-      this.environments = environments
-      this.roles = roles
-    }
-
-    if (manifest) {
-      const {
-        author,
-        changelog_url: changelogUrl,
-        description,
-        details_url: descriptionUrl,
-        icons,
-        screenshots,
-        source_url: sourceUrl,
-      }: AragonManifest = parseMetadata(manifest, 'manifest.json')
-
-      this.author = author
-      this.changelogUrl = changelogUrl
-      this.description = description
-      this.descriptionUrl = descriptionUrl
-      this.icons = icons
-      this.screenshots = screenshots
-      this.sourceUrl = sourceUrl
-    }
-
     this.address = data.address
+    this.contentUri = data.contentUri ?? undefined
```

Comment:
ES2020 🤩 wouldn't it be better to return `null`?

Reply:
Thank for the suggestions. I think in this particular case the problem is that we are implementing the `RepositoryData` interface and the `contentUri` has those types. Regardless I agree there is something a bit odd here, and I believe it's a consequence of allowing data on The Graph to be `undefined / null`. 

Anyways I like to know why you think `null` would be better than `undefine`?

Reply:
It's probably due to my mental model regarding `null` & `undefined`; for me, `null` is the proper thing to use when there's _no_ value. It indicates the absence of it. `undefined`, for me, indicates problems with fetching some data, a function that returns nothing (`void`) or some other anomalous development in processing data.

Reply:
Agreed on the mental model, I see `null` as “empty”, while `undefined` is “nothing”. I also like to keep using the same type when it makes sense, e.g. here we could use `""`, or if it was a list we could use `[]`: it removes the need to treat `null` / `undefined` as an exception for the consumer.

…I wish JS only had `null` :laughing:

Reply:
Now that you mention about using an empty object of the particular type. I was wondering if we should do that at the subgraph level and always asure that at least we have an empty object. At least for the most part of the attributes.

Reply:
> Now that you mention about using an empty object of the particular type. I was wondering if we should do that at the subgraph level and always asure that at least we have an empty object. At least for the most part of the attributes.

I think it could be nice yes. :+1: